music21/musicxml/xmlToM21.py

```diff
@@ -866,6 +867,19 @@ def xmlRootToScore(self, mxScore, inputM21=None):
             self.spannerBundle.remove(sp)
 
         s.coreElementsChanged()
+        # Fill gaps with rests where needed
+        for m in s[stream.Measure]:
+            for v in m.voices:
+                if v:  # do not bother with empty voices
+                    # the musicDataMethods use insertCore, thus the voices need to run
+                    # coreElementsChanged
+                    v.coreElementsChanged()
+                    # Fill mid-measure gaps, and find end of measure gaps by ref to measure stream
+                    # https://github.com/cuthbertlab/music21/issues/444
+                    v.makeRests(refStreamOrTimeRange=m,
+                                fillGaps=True,
+                                inPlace=True,
+                                hideRests=True)
```

Comment:
I think this is too presumptive on the part of the parser. We should be interpolating/inventing as few elements as possible that are not in the document.

Reply:
I agree! 
I just put the makeRests there because previously the parser also ran `makeRests`, but I'd be more than happy to remove it.